fineract-provider/src/main/java/org/apache/fineract/accounting/journalentry/service/JournalEntryReadPlatformServiceImpl.java

```diff
@@ -204,7 +204,12 @@ public JournalEntryData mapRow(final ResultSet rs, @SuppressWarnings("unused") f
                 }
                 Long transaction = null;
                 if (entityType != null) {
-                    transaction = Long.parseLong(transactionId.substring(1).trim());
+                    if (transactionId.substring(0, 2).equalsIgnoreCase("SH")) {
```

Comment:
This is quite the opposite of a robust implementation. If you want to fix this, let's broaden our perspective and do a proper fix. Checking the beginning of strings is definitely not the way to go. Plus, you're lacking tests anyway.